core/lib/src/fs/server.rs

```diff
@@ -237,8 +237,25 @@ impl Handler for FileServer {
                 let index = NamedFile::open(p.join("index.html")).await;
                 index.respond_to(req).or_forward((data, Status::NotFound))
             },
-            Some(p) => {
-                let file = NamedFile::open(p).await;
+            Some(mut p) => {
+                let gzip_accepted = req.headers().get("Accept-Encoding")
+                    .find(|value| value.contains("gzip"))
+                    .is_some();
+                let check_compressed = gzip_accepted && options.contains(Options::CheckCompressed);
+                if check_compressed {
+                    if let Some(file) = p.file_name() {
+                        let mut compressed = file.to_os_string();
+                        compressed.push(".gz");
+                        let compressed_file = p.with_file_name(compressed);
+                        if compressed_file.exists() {
```

Comment:
Could we use magic bytes here instead of just relying on file extensions? Could save a lot of headache if the "compressed" files are encoded wrong.

Reply:
True, though that's more disk access than just seeing if the file exists.
And I've personally seen a long history of this technique (same file name with .gz extension) for pre-gzipped content.

I figured it would be fine since we are using the extension for the mime types of the file already.

Reply:
That's fair, I'm very used to file extensions potentially not lining up with their content (working in security), but if it's the standard and hasn't caused problems so far, no harm no foul!